src/canvas/Canvas.ts

```diff
@@ -1345,17 +1344,11 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
         target.findControl(this.getViewportPoint(e));
 
     if (!corner) {
-      if ((target as Group).subTargetCheck) {
-        // hoverCursor should come from top-most subTarget,
-        // so we walk the array backwards
-        this.targets
-          .concat()
-          .reverse()
-          .map((_target) => {
-            hoverCursor = _target.hoverCursor || hoverCursor;
-          });
-      }
-      this.setCursor(hoverCursor);
+      // hoverCursor should come from top-most subTarget
+      const hoverCursor =
+        (target as Group).subTargetCheck &&
```

Comment:
should it be:
```suggestion
        (target as Group).interactive &&
```

?